### PR TITLE
Restart Qt app instance instead of process

### PR DESCRIPTION
**Pull request by nbolton**
fixes: #7896

Instead of launching a new process, which is a bit hacky, this simply starts a new Qt app instance after the existing one exits. As well as fixing a bug (where the new process hangs on exit), this also makes the process far easier to debug, since there is no new process to attach to.

I suggest we land this before: #7896

**To test:**
1. Clear settings (Help -> Clear settings)
2. Enter computer name
3. File -> Quit

Expect: Instead of the app hanging like before (I had to press ctrl+C to kill the process), it should exit cleanly.

### CODE REVIEW

File: doc/configuration.md [hunk 1]
```diff
@@ -164,5 +164,5 @@ _last = "moe:24800"
 # DESKFLOW_GUI_VERBOSE=true
 
 # Reset all settings and delete all data on startup
-# DESKFLOW_RESET_ALL=true
+# DESKFLOW_CLEAR_SETTINGS=true
```
Comment (sithlord48) [inline doc/configuration.md:167]: We need a setting to clear the setting? WHY this is a horrible pattern.

File: src/gui/src/MainWindow.cpp [hunk 2]
```diff
@@ -367,7 +367,12 @@ void MainWindow::on_m_pActionClearSettings_triggered()
 
   m_Quitting = true;
   m_SaveOnExit = false;
-  diagnostic::clearSettings(m_ConfigScopes, true);
+  diagnostic::clearSettings(m_ConfigScopes);
+
+  // Expected to cause the app to restart after exiting.
+  Q_EMIT clearedSettings();
+
+  QApplication::quit();
```
Comment (sithlord48) [inline src/gui/src/MainWindow.cpp:375]: This is the bug fix

**Review by sithlord48 — CHANGES_REQUESTED** [review]
This is alot of code to do a very simple thing. If you wanna exit when settings are cleared then tell the user you cleared the settings then exit. That's it .

File: src/gui/src/MainWindow.cpp (hunk 2 above)
Comment (nbolton) [reply src/gui/src/MainWindow.cpp:375]: I disagree, this is part of the fix.

**Comment by sithlord48** [comment]
> > If you wanna exit when settings are cleared then tell the user you cleared the settings then exit.
> 
> You're missing the point. Exiting the app, or starting a new process is horrible UX.

Then fix the settings so restart is not needed.. Cause when we do that this code is now useless...

File: doc/configuration.md (hunk 1 above)
Comment (nbolton) [reply doc/configuration.md:167]: I’d like to keep this functionality (it’s already in `master`) as it serves a specific purpose for development and debugging. Developer conveniences like this contribute directly to product quality by streamlining our workflow. This setting allows us to quickly reset all configurations without having to manually delete files or navigate through the UI.

It’s particularly valuable in testing scenarios where we need to reset the application to a 'fresh' state to verify behaviors reliably. Removing it would increase both the time and cognitive load for developers who need to clear configurations repeatedly while testing or troubleshooting. For instance, when developing the wizard.

In short, this setting boosts development efficiency and maintains a smoother workflow without impacting end users. It’s an established part of our process, adding real value with a minimal code footprint. Removing it, in my opinion, would only make things harder for the rest of us, with no tangible upside.

Comment (sithlord48) [reply doc/configuration.md:167]: If its in master that is fine we can remove it later.

**Comment by nbolton** [comment]
> Then fix the settings so restart is not needed.. Cause when we do that this code is now useless...

What are you proposing exactly? Somehow reset the state of the main window without restarting it then show the wizard?

**Comment by sithlord48** [comment]
> > Then fix the settings so restart is not needed.. Cause when we do that this code is now useless...
> 
> What are you proposing exactly? Somehow reset the state of the main window without restarting it then show the wizard?

Yes.
